diff --git a/.github/workflows/docgen.yml b/.github/workflows/docgen.yml
index 593c4e90f..8175a8dfe 100644
--- a/.github/workflows/docgen.yml
+++ b/.github/workflows/docgen.yml
@@ -17,7 +17,7 @@ jobs:
       image: docker.io/pytorch/manylinux-builder:cuda12.4
       options: --gpus all
     env:
-      CUDA_HOME: /usr/local/cuda-12.4
+      CUDA_HOME: /usr/local/cuda
       VERSION_SUFFIX: cu124
       CU_VERSION: cu124
       CHANNEL: nightly
diff --git a/MODULE.bazel b/MODULE.bazel
index add7821fc..d6b781cb6 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -36,7 +36,7 @@ new_local_repository = use_repo_rule("@bazel_tools//tools/build_defs/repo:local.
 new_local_repository(
     name = "cuda",
     build_file = "@//third_party/cuda:BUILD",
-    path = "/usr/local/cuda-12.4/",
+    path = "/usr/local/cuda/",
 )
 
 new_local_repository(
diff --git a/core/util/Exception.h b/core/util/Exception.h
index 872560146..87bedfb34 100644
--- a/core/util/Exception.h
+++ b/core/util/Exception.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <exception>
 #include <sstream>
 #include <string>
diff --git a/py/ci/build_whl.sh b/py/ci/build_whl.sh
index d8f91da68..f732e1b6a 100755
--- a/py/ci/build_whl.sh
+++ b/py/ci/build_whl.sh
@@ -3,15 +3,15 @@
 # Example usage: docker run -it -v$(pwd):/workspace/TensorRT build_torch_tensorrt_wheel /bin/bash /workspace/TensorRT/py/ci/build_whl.sh
 
 export CXX=g++
-export CUDA_HOME=/usr/local/cuda-12.4
+export CUDA_HOME=/usr/local/cuda
 export PROJECT_DIR=/workspace/TensorRT
 
 rm -rf /usr/local/cuda
 
-if [[ $CUDA_HOME == "/usr/local/cuda-12.4" ]]; then
+if [[ $CUDA_HOME == "/usr/local/cuda" ]]; then
     cp -r /usr/local/cuda-11.8 /usr/local/cuda
     cp -r /usr/local/cuda-12.0/ /usr/local/cuda/
-    rsync -a /usr/local/cuda-12.4/ /usr/local/cuda/
+    rsync -a /usr/local/cuda/ /usr/local/cuda/
     export CUDA_HOME=/usr/local/cuda
 else
     ln -s $CUDA_HOME /usr/local/cuda
diff --git a/py/torch_tensorrt/__init__.py b/py/torch_tensorrt/__init__.py
index d7f5e7ba5..0cb0cc2ed 100644
--- a/py/torch_tensorrt/__init__.py
+++ b/py/torch_tensorrt/__init__.py
@@ -56,7 +56,7 @@ except ImportError:
             ctypes.CDLL(_find_lib(lib, WIN_PATHS))
 
     elif sys.platform.startswith("linux"):
-        LINUX_PATHS = ["/usr/local/cuda-12.4/lib64", "/usr/lib", "/usr/lib64"]
+        LINUX_PATHS = ["/usr/local/cuda/lib", "/usr/lib", "/usr/lib64"]
         if "LD_LIBRARY_PATH" in os.environ:
             LINUX_PATHS += os.environ["LD_LIBRARY_PATH"].split(os.path.pathsep)
 
diff --git a/toolchains/legacy/WORKSPACE.x86_64 b/toolchains/legacy/WORKSPACE.x86_64
index 1428ec439..0bbeaeb82 100644
--- a/toolchains/legacy/WORKSPACE.x86_64
+++ b/toolchains/legacy/WORKSPACE.x86_64
@@ -43,7 +43,7 @@ local_repository(
 new_local_repository(
     name = "cuda",
     build_file = "@//third_party/cuda:BUILD",
-    path = "/usr/local/cuda-12.4/",
+    path = "/usr/local/cuda/",
 )
 
 new_local_repository(
diff --git a/toolchains/legacy/WORKSPACE.x86_64.release.ubuntu b/toolchains/legacy/WORKSPACE.x86_64.release.ubuntu
index e7bad3d30..47b837056 100644
--- a/toolchains/legacy/WORKSPACE.x86_64.release.ubuntu
+++ b/toolchains/legacy/WORKSPACE.x86_64.release.ubuntu
@@ -43,7 +43,7 @@ local_repository(
 new_local_repository(
     name = "cuda",
     build_file = "@//third_party/cuda:BUILD",
-    path = "/usr/local/cuda-12.4",
+    path = "/usr/local/cuda",
 )
 
 new_local_repository(
